lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.Common.cpp

```diff
@@ -67,11 +68,20 @@ namespace PlatformAgnostic
                     num = newNum;
 
                     str++;
+                    numLength++;
                 }
 
+                *len += numLength;
```

Comment:
*len is accumulating because the string may contain other digits or characters before that, so `=` would just overwrite the value and would lead to unnecessary iterations overall.